auditlog-fhir-server/src/main/java/ca/uhn/fhir/jpa/starter/MyConsentService.java

```diff
@@ -155,13 +166,43 @@ private void createAuditEvent(RequestDetails theRequestDetails, boolean wasSucce
     myAuditEventDao.create(auditEvent);
   }
 
-  //Array toString with removed [brackets] so the result looks like: value1,value2,value3
-  private String toCommaSeparatedStringList(String[] theInputStringArray) {
-    if (theInputStringArray.length > 0) {
-      String commaSeparatedListWithBrackets = Arrays.toString(theInputStringArray);
-      return commaSeparatedListWithBrackets.substring(1, commaSeparatedListWithBrackets.length() - 1);
+  //retrieve the first subject reference id (patient id) of a resource (not exhaustive, only for resources of the tested radiological workflow)
+  private String retrieveSubjectIdOrPatientId(RequestDetails theRequestDetails) {
+    //TODO: figure out how to transer caseId for the operation  http://endpoint/DiagnosticReport/$DiagnosticReportId/$fhirToCDA
+
+    // GET request
+    if (theRequestDetails.getRequestType().equals("GET")) {
+      //GET request that contains a subject parameter
+      if (theRequestDetails.getParameters() != null &&
+        !theRequestDetails.getParameters().isEmpty() &&
+        theRequestDetails.getParameters().get("subject") != null) {
+        return theRequestDetails.getParameters().get("subject")[0];
+        //GET request that contains a patient's id
+      } else if (theRequestDetails.getResourceName().equals("Patient") &&
+        theRequestDetails.getId() != null &&
+        theRequestDetails.getId().getValue() != null &&
+        !theRequestDetails.getId().getValue().isEmpty()) {
+        return theRequestDetails.getId().getValue();
+      }
+    }
+
+    //PUT, POST, DELETE... or even GET request as long as it does not contain a parameter "subject"
+    if (theRequestDetails.getResourceName().equals("Patient")) {
+      return theRequestDetails.getId().getValue();
+    } else if (theRequestDetails.getResourceName().equals("Appointment")) {
+      Appointment appointment = (Appointment) theRequestDetails.getResource();
+      return appointment.getParticipantFirstRep().getActor().getReference();
+    } else if (theRequestDetails.getResourceName().equals("Procedure")) {
+      Procedure procedure = (Procedure) theRequestDetails.getResource();
+      return procedure.getSubject().getReference();
+    } else if (theRequestDetails.getResourceName().equals("Media")) {
+      Media media = (Media) theRequestDetails.getResource();
+      return media.getSubject().getReference();
+    } else if (theRequestDetails.getResourceName().equals("DiagnosticReport")) {
+      DiagnosticReport diagnosticReport = (DiagnosticReport) theRequestDetails.getResource();
+      return diagnosticReport.getSubject().getReference();
     } else {
-      return "";
+      throw new NotImplementedException("There is no existing implementation yet to retrieve a subject for the resource " + theRequestDetails.getResourceName() + ".");
```

Comment:
Just as I note, as I have seen this quite often from other programmers, too:
NotImplementedException is an exception that is of a specific framework in Java, whereas in C# it is from the standard library. The corresponding standard Exception in Java would be UnsupportedOperationException. You do not have to change that, I just wanted to let you know.

Reply:
thanks, never knew that even though i have noticed a weird namespace. will change that.